app/src/org/commcare/fragments/CommCarePreferenceFragment.java

```diff
@@ -0,0 +1,131 @@
+        implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+    private static final String TAG = CommCarePreferenceFragment.class.getSimpleName();
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        GoogleAnalyticsUtils.reportPrefActivityEntry(GoogleAnalyticsFields.CATEGORY_CC_PREFS);
+        setupUI();
+        initPrefs();
+        loadPrefs();
+    }
+
+    @CallSuper
+    protected void initPrefs() {
+        if(isAppLevelPreference()) {
+            PreferenceManager prefMgr = getPreferenceManager();
+            prefMgr.setSharedPreferencesName((CommCareApplication.instance().getCurrentApp().getPreferencesFilename()));
+        }
+    }
+
+    @CallSuper
+    protected void loadPrefs() {
+        // Add 'general' preferences, defined in the XML file
+        addPreferencesFromResource(getPreferencesResource());
+
+        GoogleAnalyticsUtils.createPreferenceOnClickListeners(getPreferenceManager(), getPrefKeyAnalyticsEventMap(),
+                GoogleAnalyticsFields.CATEGORY_CC_PREFS);
+        setupPrefClickListeners();
+        setupLocalizedText();
+    }
+
+    @CallSuper
+    protected void setupUI() {
+        if (getActivity() != null) {
+            getActivity().setTitle(getTitle());
+        }
+    }
+
+    private void setupLocalizedText() {
+        Map<String, String> prefToTitleMap = getPrefKeyTitleMap();
+        if (prefToTitleMap != null) {
+            PreferenceScreen screen = getPreferenceScreen();
+            for (int i = 0; i < screen.getPreferenceCount(); i++) {
+                String key = screen.getPreference(i).getKey();
+                if (prefToTitleMap.containsKey(key)) {
+                    try {
+                        String localizedString = Localization.get(prefToTitleMap.get(key));
+                        screen.getPreference(i).setTitle(localizedString);
+                    } catch (NoLocalizedTextException nle) {
+                        Log.w(TAG, "Unable to localize: " + prefToTitleMap.get(key));
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // register the preference change listener
+        getPreferenceScreen().getSharedPreferences()
+                .registerOnSharedPreferenceChangeListener(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        // unregister the preference change listener
+        getPreferenceScreen().getSharedPreferences()
+                .unregisterOnSharedPreferenceChangeListener(this);
+    }
+
+    /**
+     * Use this to reload prefs screen
+     */
+    @CallSuper
+    protected void reset() {
+        getPreferenceScreen().removeAll();
+        loadPrefs();
+    }
+
+    /**
+     *
+     * @return whether the preference should be stored in app specific preference file.
+     */
+    protected boolean isAppLevelPreference() {
```

Comment:
Can we call this `isPersistentAppPreference()` instead? There are some preference fragments that relate to app-level actions/settings, but return `false` for this method because we don't want to store them in a persistent prefs file